### ISSUE TITLE
stack followed by transpose fails

### ISSUE THREAD

**Issue by shoyer** [ContinuumIO/dask#100]
```python
import numpy as np
import dask
import dask.array as da

da.set_options(get=dask.get)
x = da.from_array(np.array([1]), blockshape=(1,))
da.stack([x]).transpose().compute()
```
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-14-2da89db80151> in <module>()
      5 da.set_options(get=dask.get)
      6 x = da.from_array(np.array([1]), blockshape=(1,))
----> 7 da.stack([x]).transpose().compute()

/Users/shoyer/dev/dask/dask/array/core.py in compute(self, **kwargs)
    595     @wraps(compute)
    596     def compute(self, **kwargs):
--> 597         return compute(self, **kwargs)
    598 
    599     __float__ = __int__ = __bool__ = __complex__ = compute

/Users/shoyer/dev/dask/dask/array/core.py in compute(*args, **kwargs)
    430     dsk = merge(*[arg.dask for arg in args])
    431     keys = [arg._keys() for arg in args]
--> 432     results = get(dsk, keys, **kwargs)
    433 
    434     results2 = [rec_concatenate(x) if arg.shape else unpack_singleton(x)

/Users/shoyer/dev/dask/dask/array/core.py in get(dsk, keys, get, **kwargs)
    823     dsk3 = remove_full_slices(dsk2)
    824     dsk4 = inline_functions(dsk3, fast_functions=fast_functions)
--> 825     return get(dsk4, keys, **kwargs)
    826 
    827 

/Users/shoyer/dev/dask/dask/core.pyc in get(d, key, get, concrete, **kwargs)
    111         v = (get(d, k, get=get, concrete=concrete) for k in key)
    112         if concrete:
--> 113             v = list(v)
    114     elif ishashable(key) and key in d:
    115         v = d[key]

/Users/shoyer/dev/dask/dask/core.pyc in <genexpr>((k,))
    109     get = get or _get
    110     if isinstance(key, list):
--> 111         v = (get(d, k, get=get, concrete=concrete) for k in key)
    112         if concrete:
    113             v = list(v)

/Users/shoyer/dev/dask/dask/core.pyc in get(d, key, get, concrete, **kwargs)
    111         v = (get(d, k, get=get, concrete=concrete) for k in key)
    112         if concrete:
--> 113             v = list(v)
    114     elif ishashable(key) and key in d:
    115         v = d[key]

/Users/shoyer/dev/dask/dask/core.pyc in <genexpr>((k,))
    109     get = get or _get
    110     if isinstance(key, list):
--> 111         v = (get(d, k, get=get, concrete=concrete) for k in key)
    112         if concrete:
    113             v = list(v)

/Users/shoyer/dev/dask/dask/core.pyc in get(d, key, get, concrete, **kwargs)
    111         v = (get(d, k, get=get, concrete=concrete) for k in key)
    112         if concrete:
--> 113             v = list(v)
    114     elif ishashable(key) and key in d:
    115         v = d[key]

/Users/shoyer/dev/dask/dask/core.pyc in <genexpr>((k,))
    109     get = get or _get
    110     if isinstance(key, list):
--> 111         v = (get(d, k, get=get, concrete=concrete) for k in key)
    112         if concrete:
    113             v = list(v)

/Users/shoyer/dev/dask/dask/core.pyc in get(d, key, get, concrete, **kwargs)
    122         if get is _get:
    123             # use non-recursive method by default
--> 124             return _get_task(d, v)
    125         func, args = v[0], v[1:]
    126         args2 = [get(d, arg, get=get, concrete=False) for arg in args]

/Users/shoyer/dev/dask/dask/core.pyc in _get_task(d, task, maxdepth)
     59         func, args, results = stack[-1]
     60         if not args:
---> 61             val = func(*results)
     62             if len(stack) == 1:
     63                 return val

/Users/shoyer/miniconda/envs/xray-dev/lib/python2.7/site-packages/toolz/functoolz.pyc in __call__(self, *args, **kwargs)
    214     def __call__(self, *args, **kwargs):
    215         try:
--> 216             return self._partial(*args, **kwargs)
    217         except TypeError:
    218             # If there was a genuine TypeError

/Users/shoyer/miniconda/envs/xray-dev/lib/python2.7/site-packages/numpy/core/fromnumeric.pyc in transpose(a, axes)
    535     except AttributeError:
    536         return _wrapit(a, 'transpose', axes)
--> 537     return transpose(axes)
    538 
    539 

ValueError: axes don't match array
```

**Comment by mrocklin** [comment]
OK, definitely an issue.  It appears that it thinks the shape of `da.stack([x]) == (1, 1)`

**Comment by mrocklin** [comment]
Oh, perhaps it should be.  The individual blocks presumably need to be expanded out along the new dimension.

**Comment by mrocklin** [comment]
Fixed in #101 

```python
In [1]: import numpy as np

In [2]: import dask

In [3]: import dask.array as da

In [4]: 

In [4]: da.set_options(get=dask.get)
Out[4]: <dask.context.set_options at 0x7f4bc9064ad0>

In [5]: x = da.from_array(np.array([1]), blockshape=(1,))

In [6]: da.stack([x]).transpose().compute()
Out[6]: array([[1]])
```
